strawberry/relay/types.py

```diff
@@ -0,0 +1,706 @@
+import abc
+import dataclasses
+import inspect
+import itertools
+import sys
+import uuid
+from typing import (
+    Any,
+    Awaitable,
+    ClassVar,
+    Dict,
+    Generic,
+    Iterable,
+    List,
+    Optional,
+    Sequence,
+    Sized,
+    Tuple,
+    Type,
+    TypeVar,
+    Union,
+    cast,
+    overload,
+)
+from typing_extensions import Literal, get_args, get_origin
+
+from strawberry.field import field
+from strawberry.lazy_type import LazyType
+from strawberry.object_type import interface, type
+from strawberry.scalars import ID
+from strawberry.type import StrawberryContainer
+from strawberry.types.info import Info
+from strawberry.types.types import TypeDefinition
+from strawberry.utils.await_maybe import AwaitableOrValue
+
+from .utils import from_base64, to_base64
+
+_T = TypeVar("_T")
+_R = TypeVar("_R")
+connection_typename = "arrayconnection"
+
+NodeType = TypeVar("NodeType", bound="Node")
```

Comment:
This looks like a class name to me, and reading the docs example for custom connections had me really confused (wrgt how it was implemented, not how to use it).

I'm not sure how to improve this really, though. Perhaps `TNode` does a better job at hinting that this is a generic type var?

however, looking at some more places this is used, I think `NodeType` works really well, so perhaps the doc mentioned is the only confusing part.. I'll ponder on this a bit more.

Reply:
I'm open to change it to `TNode` or anything else that makes more sense. I have to admit that I'm not good at naming things (`strawberry-django-plus` is a good example of that hahaha)

Reply:
Naming stuff is among the top 3 most difficult programming jobs! :P 

I think I landed in being in favour of keeping it as `NodeType`, left the comment regardless to be transparent about my review process.